### PR TITLE
Feature/refine goal seek

### PR DESCRIPTION
**Pull request by seb231**
I needed to be able to pull out the current result for a particular state interactively in order to calculate population increase and so broke up a fn into two

### CODE REVIEW

**Comment by seb231** [comment]
Docstring placement needs fixing

File: src/clj/witan/send/goal_seek.clj [hunk 1]
```diff
@@ -55,18 +67,61 @@
 (defn target-pop-exceeded? [pop target-pop]
   (> pop (apply max target-pop)))
 
-(defn target-results [m start end step base-config target]
-  "Takes a map of keys partially matching a transition, a start, end and step range to modify
-   the transition by, a template config and a map containing a target population range and year
+(defn assoc-transition-params [config m]
+  (let [state (create-transition-modifier-seq m 1 2 1)]
```
Comment (otfrom) [inline src/clj/witan/send/goal_seek.clj]: worth putting in an example of what the before an dafter states of the config are here?

File: src/clj/witan/send/goal_seek.clj [hunk 2]
```diff
@@ -55,18 +67,61 @@
...
+    (-> config
+        (assoc-in (first (first state))
+                  (first (second (first state)))))))
+
+(defn target-result [config target-year & m]
+  "Takes a baseline config to use as a template, a target year and an optional map of keys
+   partially matching a transition, and a value to modify by"
```
Comment (otfrom) [inline src/clj/witan/send/goal_seek.clj]: fix docstring

File: src/clj/witan/send/goal_seek.clj [hunk 3]
```diff
@@ -55,18 +67,61 @@
...
+                     first
+                     :modify-transition-by)
+        result (do (main/run-recorded-send config)
+                   (get-target-pop (state-pop config)))
+        current-pop (get-current-pop target-year result)
+        achieved-pop (->> result
+                          (filter #(= (:year %) target-year))
+                          first
+                          :population)]
+    (do (println "Modifier:" modifier)
```
Comment (otfrom) [inline src/clj/witan/send/goal_seek.clj]: you don't need do here

File: src/clj/witan/send/goal_seek.clj [hunk 4]
```diff
@@ -55,18 +67,61 @@
...
-          result (do (main/run-recorded-send config)
-                     (get-target-pop config))
-          current-pop (get-current-pop (:year target) result)
-          diff (pop-diff-by-year (:year target) result)]
-      (if (target-pop-exceeded? current-pop (:population target))
-        (println "Population already exceeds target population")
-        (when-not (within-pop-range? (:population target) diff)
-          (recur rest-configs))))))
+(defn target-results [base-config target m & step]
+  "Takes a baseline config to use as a template, a map containing a target population
```
Comment (otfrom) [inline src/clj/witan/send/goal_seek.clj]: fix docstring

File: src/clj/witan/send/model/prepare.clj [hunk 5]
```diff
@@ -184,7 +184,52 @@
                                             (op2 (:need-2 %) val2)))))))
 
 (defn test-predicates [data pred-map]
-  (map (fn [[k v]] (= (k data) v)) pred-map))
+  "data can be any map, while pred-map must be a sequence of key-value pairs
+   matching the key to filter on and the value to filter by"
```
Comment (seb231) [inline src/clj/witan/send/model/prepare.clj]: fix docstring

**Pull request merged**